simulations/vip-123/vip-123/simulations.ts

```diff
@@ -0,0 +1,295 @@
+  },
+  {
+    name: "vMATIC",
+    address: "0x5c9476FcD6a4F9a3654139721c949c2233bBbBc8",
+    price: "0.886519",
+    assetAddress: "0xCC42724C6683B7E57334c4E856f4c9965ED682bD",
+    feed: "0x7ca57b0ca6367191c94c8914d7df09a57655905f",
+  },
+  {
+    name: "vCAKE",
+    address: "0x86aC3974e2BD0d60825230fa6F355fF11409df5c",
+    price: "1.55885968",
+    assetAddress: "0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82",
+    feed: "0xb6064ed41d4f67e353768aa239ca86f4f73665a1",
+  },
+  {
+    name: "vAAVE",
+    address: "0x26DA28954763B92139ED49283625ceCAf52C6f94",
+    price: "63.327181",
+    assetAddress: "0xfb6115445Bff7b52FeB98650C87f44907E58f802",
+    feed: "0xa8357bf572460fc40f4b0acacbb2a6a61c89f475",
+  },
+  {
+    name: "vTUSD",
+    address: "0x08CEB3F4a7ed3500cA0982bcd0FC7816688084c3",
+    price: "0.99965988",
+    assetAddress: "0x14016E85a25aeb13065688cAFB43044C2ef86784",
+    feed: "0xa3334a9762090e827413a7495afece76f41dfc06",
+  },
+  {
+    name: "vTRX",
+    address: "0x61eDcFe8Dd6bA3c891CB9bEc2dc7657B3B422E93",
+    price: "0.07737686",
+    assetAddress: "0x85EAC5Ac2F758618dFa09bDbe0cf174e7d574D5B",
+    feed: "0xf4c5e535756d11994fcbb12ba8add0192d9b88be",
+  },
+  {
+    name: "vTRX",
+    address: "0xC5D3466aA484B040eE977073fcF337f2c00071c1",
+    price: "77376860000",
+    assetAddress: "0xCE7de646e7208a4Ef112cb6ed5038FA6cC6b12e3",
+    feed: "0xf4c5e535756d11994fcbb12ba8add0192d9b88be",
+  },
+  {
+    name: "VAI",
+    address: "0x4BD17003473389A42DAF6a0a729f6Fdb328BbBd7",
+    price: "0.9850251",
+    assetAddress: "0x4BD17003473389A42DAF6a0a729f6Fdb328BbBd7",
+    feed: "0x058316f8Bb13aCD442ee7A216C7b60CFB4Ea1B53",
+  },
+];
+
+const directVTokens: DirectVTokenConfig[] = [
+  {
+    name: "LUNA",
+    address: "0xb91A659E88B51474767CD97EF3196A3e7cEDD2c8",
+    price: "0.000001",
+  },
+  {
+    name: "UST",
+    address: "0x78366446547D062f45b4C0f320cDaa6d710D87bb",
+    price: "0.000001",
+  },
+  {
+    name: "CAN",
+    address: "0xeBD0070237a0713E8D94fEf1B728d3d993d290ef",
+    price: "0.000000000000000001",
+  },
+];
+
+forking(28526142, () => {
+  const provider = ethers.provider;
+
+  describe("Pre-VIP behavior", async () => {
+    let priceOracle: ethers.Contract;
+
+    before(async () => {
+      priceOracle = new ethers.Contract(PRICE_ORACLE, PRICE_ORACLE_ABI, provider);
+      await setMaxStalePeriodInOracle(COMPTROLLER);
+    });
+
+    it("validate vToken prices", async () => {
+      for (let i = 0; i < vTokens.length; i++) {
+        const vToken = vTokens[i];
+        const price = await priceOracle.getUnderlyingPrice(vToken.address);
+        expect(price).to.be.equal(parseUnits(vToken.price, 18));
+      }
+
+      for (let i = 0; i < directVTokens.length; i++) {
+        const vToken = directVTokens[i];
+        const price = await priceOracle.getUnderlyingPrice(vToken.address);
+        expect(price).to.be.equal(parseUnits(vToken.price, 18));
+      }
+    });
```

Comment:
We can register a separate test for each token, this way it would be straightforward to debug if some of the prices is invalid:

```suggestion
    vTokens.forEach((vToken: VTokenConfig) => {
      it(`returns Chainlink price for ${vToken.name}`, async () => {
        const price = await priceOracle.getUnderlyingPrice(vToken.address);
        expect(price).to.be.equal(parseUnits(vToken.price, 18));
      });
    });

    directVTokens.forEach((vToken: DirectVTokenConfig) => {
      it(`returns hardcoded price for ${vToken.name}`, async () => {
        const price = await priceOracle.getUnderlyingPrice(vToken.address);
        expect(price).to.be.equal(parseUnits(vToken.price, 18));
      });
    });
```